resqpy/well/_blocked_well.py

```diff
@@ -575,6 +575,38 @@ def set_for_column(self, well_name, grid, col_ji0, skip_inactive = True, length_
 
         return self.derive_from_dataframe(df, self.well_name, grid, use_face_centres = True, length_uom = length_uom)
 
+    def dtop_dbot_to_layers(self, df, grid):
+        """Populates the wellspec dataframe with corresponding L when DTOP and DBOT are specified but L is not
+        arguments
+          df (pd.DataFrame): Dataframe of wellspec information
+          grid (grid.Grid object): Grid where well is located
+        returns
+          df_expanded (pd.DataFrame): DataFrame of wellspec information where L is populated with the layers corresponding to the DTOP and DBOT depths
+        """
+        corp = grid.corner_points()
+        nk,nj,ni = corp.shape[:3]
+        df_expanded = []
+
+        for ix,data in df.iterrows():
+            if np.isnan(data['L']):
+                dtop = data['DTOP']
+                dbot = data['DBOT']
+                j,i = int(data['JW']), int(data['IW'])
+                for k in range(nk):
+                    if dbot == 'BOT':
```

Comment:
does DTOP need to be checked when 'BOT' is in use for DBOT? (I haven't looked in the manual!)